.github/workflows/ci-on-comment.yaml

```diff
@@ -0,0 +1,123 @@
+name: CI
+on:
+  issue_comment:
+    types: [created]
+
+jobs:
+  kata-containers-ci-on-comment:
```

Comment:
Disclaimer: my knowledge with gh actions is limited, I've often forgot some details and have to consult the docs.

According to the docs https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#issue_comment an `issue_comment` is triggered for comments either in an issue or pull request.

Here we are interested on comments in pull requests. So could we conditionally run the `kata-containers-ci-on-comment` only and only if `github.event.issue.pull_request` to avoid unnecessary runs? If yes, then the `if` clause in `Start CI` should be refactored.

BTW, I think this will fix a bug...because the way it is implemented, IIUC, if I a send a comment to an issue then all the steps, except for `Start CI`, are going to run. And it does not make sense to run, say, `Fetch CI run ID` if `Start CI` is not running.

Reply:
You're 100% right, will put a condition on the entire job!

Reply:
Scoped the condition to entire job, PTAL

Reply:
Cool, thanks!